### DISCUSSION TITLE
How to make if checks inside set((state))?

### DISCUSSION THREAD

**Discussion by russ0133** [pmndrs/zustand#1301] [Q&A]
The code should be self explanatory, but I have a set function in zustand that takes 2 arguments (property: string, overrideData: number) and I need to make if checks on the first one to make sure if incorrect data is sent, then nothing will be set.

I'm doing the following: (works fine)
![image](https://user-images.githubusercontent.com/26175960/190888797-4e46efb7-7eeb-4a11-b0eb-f7bc59af56d2.png)

And trying to do the following: (all hell breaks loose, typescript shows errors on the whole file)
![image](https://user-images.githubusercontent.com/26175960/190888844-0a1b6eae-c733-404a-9c3b-aaa229bd91a7.png)

I reckon I can just make another function inside the store, which makes all the checks and then calls setPropertyData, but that doesn't seem right to me, and then again, how would I access the state outside of a set((state))?

**Comment by dai-shi** [comment]
You can `return state` to make it NO-OP.
